daemons/controld/controld_execd.c

```diff
@@ -1978,7 +1978,41 @@ construct_op(lrm_state_t * lrm_state, xmlNode * rsc_op, const char *rsc_id,
             resolve_versioned_attrs(lrm_state, rsc, rsc_op, params);
         }
 #endif
+
+        /* pcmk_monitor_timeout overrides CRM_meta_timeout for stonith
+        * start/monitor
+        */
+        if (safe_str_eq(rsc->standard, PCMK_RESOURCE_CLASS_STONITH)
+            && pcmk__str_any_of(op->op_type, CRMD_ACTION_START,
+                                CRMD_ACTION_STATUS, NULL)) {
```

Comment:
The thing about monitor using `CRM_meta_timeout` in `construct_op()` is that if it **does** time out after say, 60 seconds, the "Result of..." message says that it timed out after 20 seconds.

Reply:
maybe replace it only if it's greater

Reply:
although there are scenarios where even that's not correct

CRM_meta_timeout=20s, pcmk_monitor_timeout=60s: 1 try timing out at 60s
CRM_meta_timeout=120s, pcmk_monitor_timeout=60s: 2 tries timing out at 120s
CRM_meta_timeout=120s, pcmk_monitor_timeout=60s, pcmk_monitior_retries=1: 1 try timing out at 60s
CRM_meta_timeout=180s, pcmk_monitor_timeout=60s: 2 tries timing out at 120s

we could just change the log message to say initial_timeout, to give a hint that it could have changed at some point (though that's only true of stonith operations)